File: snippets/go/v0_11/concurrency/keys.go
package concurrency

import (
	"context"

	"github.com/inngest/inngest/pkg/enums"
	"github.com/inngest/inngestgo"
)

func KeysSnippet() {
	client, _ := inngestgo.NewClient(inngestgo.ClientOpts{AppID: "my-app"})

	// !snippet:start
	_, err := inngestgo.CreateFunction(
		client,
		inngestgo.FunctionOpts{
			Name: "generate-ai-summary",
			Concurrency: []inngestgo.ConfigStepConcurrency{
				{
					Scope: enums.ConcurrencyScopeFn,
					Key:   inngestgo.StrPtr("event.data.account_id"),
					Limit: 10,
				},
			},
		},
		inngestgo.EventTrigger("ai/summary.requested", nil),
		func(ctx context.Context, input inngestgo.Input[map[string]any]) (any, error) {
			// Your function handler here
			return nil, nil
		},
	)
	// !snippet:end

	_ = err
}

File: snippets/py/v0_5/concurrency/keys.py
import inngest

inngest_client = inngest.Inngest(app_id="my-app")


# !snippet:start
@inngest_client.create_function(
    fn_id="generate-ai-summary",
    concurrency=[
        inngest.Concurrency(
            scope="fn",
            key="event.data.account_id",
            limit=10,
        )
    ],
    trigger=inngest.TriggerEvent(event="ai/summary.requested"),
)
async def generate_ai_summary(ctx: inngest.Context):
    # Your function handler here
    pass
